extensions/guacamole-auth-restrict/src/main/java/org/apache/guacamole/auth/restrict/RestrictionVerificationService.java

```diff
@@ -0,0 +1,337 @@
+                if (restriction.appliesNow())
+                    return false;
+            }
+        }
+        
+        // If no allowed entries are present, return true, allowing the login
+        // or connection to continue.
+        if (allowedTimeString == null || allowedTimeString.isEmpty())
+            return true;
+        
+        List<DailyRestriction> allowedTimes = 
+                TimeRestrictionParser.parseString(allowedTimeString);
+        
+        // Allowed entries are present, loop through them and check for a valid time.
+        for (DailyRestriction restriction : allowedTimes) {
+            // If this time allows the login or connection return true.
+            if (restriction.appliesNow())
+                return true;
+        }
+        
+        // We have allowed entries, but login hasn't matched, so deny it.
+        return false;
+        
+    }
+    
+    /**
+     * Given the strings of allowed and denied hosts, verify that the login or
+     * connection should be allowed from the given remote address. If the action
+     * should not be allowed, return false - otherwise, return true.
+     * 
+     * @param allowedHostsString
+     *     The string containing a semicolon-separated list of hosts from
+     *     which the login or connection should be allowed, or null or an empty
+     *     string if no specific set of allowed hosts is defined.
+     * 
+     * @param deniedHostsString
+     *     The string containing a semicolon-separated list of hosts from
+     *     which the login or connection should be denied, or null or an empty
+     *     string if no specific set of denied hosts is defined.
+     * 
+     * @param remoteAddress
+     *     The IP address from which the user is logging in or has logged in
+     *     and is attempting to connect from, if it is known. If it is unknown
+     *     and restrictions are defined, the login or connection will be denied.
+     * 
+     * @return
+     *     True if the login or connection should be allowed by the host-based
+     *     restrictions, otherwise false.
+     */
+    private static boolean allowedByHostRestrictions(String allowedHostsString,
+            String deniedHostsString, String remoteAddress) {
+        
+        HostName remoteHostName = new HostName(remoteAddress);
+        
+        // If attributes do not exist or are empty then the action is allowed.
+        if ((allowedHostsString == null || allowedHostsString.isEmpty()) 
+                && (deniedHostsString == null || deniedHostsString.isEmpty()))
+            return true;
+        
+        // If the remote address cannot be determined, and restrictions are
+        // in effect, log an error and deny the action.
+        if (remoteAddress == null || remoteAddress.isEmpty()) {
+            LOGGER.error("Host-based restrictions are present, but the remote address is invalid or could not be resolved.");
```

Comment:
`error` seems like a bit overkill for a case of something that we expected might be not present not being present. Seems like this should be `warning` at worst, or maybe even `info` if not knowing the remote address is really expected.

Reply:
I'm happy to make this `warn` or `info` - my rationale for going with `error` is that, under this condition, the login will actually be denied, which seems to make it an error condition.

Reply:
Okay, I changed this to `warn`, after reviewing some of the other Guacamole Client code and seeing what was used in other cases.